Asynctask And Retrofit requests

tolmadima opened the pull request:
Случайно описал все что сделал в коммите.

tolmadima committed:
Added setters

tolmadima committed:
Changed to prefix

tolmadima committed:
Fixed values of consts

tolmadima committed:
Changed view to viewNews

tolmadima committed:
Cleaning of code

tolmadima committed:
Update MainActivity.java

tolmadima committed:
Create AsyncTaskRequest.java

tolmadima committed:
Tried to use asynctask

tolmadima committed:
Trying work with AsyncTask

tolmadima committed:
Работа с запросами

Поправил запрос АсинкТаска, попытка работы с ретрофитом, не могу реализовать вывод джейсон массива в обычный с целью вывода в асинктаск. На ретрофит не хватило времени,сам запрос сделан еще давно, но не доделан потому что не разобрался с ПОДЖО файлом, может быть завтра днем еще что-то сделаю, если будет время перед встречей.

ashvidchenko commented on app/src/main/java/com/example/lastfm/Post.java:
```diff
@@ -0,0 +1,39 @@
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.SerializedName;
+
+public class Post {
```
The name of the class doesn't match its content. Rename it.

tolmadima replied:
Deleted.

ashvidchenko commented on app/src/main/java/com/example/lastfm/JSONPlaceHolderApi.java:
```diff
@@ -0,0 +1,24 @@
+    public Call<List<Post>> getPostOfUser(@Query("userId") int id);
+
+    @POST("/posts")
+    public Call<Post> postData(@Body Post data);
```
Why do u need all 4 methods?

tolmadima replied:
Deleted.

ashvidchenko commented on app/src/main/java/com/example/lastfm/MainActivity.java:
```diff
@@ -4,81 +4,128 @@
+        new GetTopTask(textView).execute(url);
+    }
+
+    private static class GetTopTask extends AsyncTask<String, Void, String> {
```
Top of what?

tolmadima replied:
Changed to GetTopArtistTask

ashvidchenko commented on app/src/main/java/com/example/lastfm/MainActivity.java:
```diff
@@ -4,81 +4,128 @@
-    private RecyclerView newsRecyclerView;
-    private NewsAdapter newsAdapter;
 
+    private static final String APP_ID = "b4ab3bf82dcb495e182e04cfc1f12b7b";
```
Why do u need this variable? Why is it static?

tolmadima replied:
Changed

tolmadima replied:
Needed to set APP_ID in get request

ashvidchenko replied:
move this variable to the resource file

ashvidchenko commented on app/src/main/java/com/example/lastfm/MainActivity.java:
```diff
@@ -4,81 +4,128 @@
 
         initRecyclerView();
+        loadArtists();
+        TextView textView = findViewById(R.id.asynctask_view);
```
Why did u name the resource with the name "asynctask_view"?

tolmadima replied:
That was test TextView, deleted

ashvidchenko commented on app/src/main/java/com/example/lastfm/MainActivity.java:
```diff
@@ -4,81 +4,128 @@
+        }
+        @Override
+        protected String doInBackground(String... strings) {
+            String artistName = "UNDEFINED";
```
Why do u need it?

tolmadima replied:
Was needed to see changes of artistName

tolmadima replied:
Deleted.

ashvidchenko commented on app/src/main/java/com/example/lastfm/MainActivity.java:
```diff
@@ -4,81 +4,128 @@
 
-
-        private Collection<NewsData> getNews(){
+        private Collection<NewsData> getArtists(){
```
the method "getArtists" must not return a collection of objects NewsData. And it's better to use List instead of Collection

tolmadima replied:
Changed to ArrayList

ashvidchenko replied:
u didn't change the returned type

tolmadima replied:
This string is deleted.

ashvidchenko commented on app/src/main/java/com/example/lastfm/MainActivity.java:
```diff
@@ -4,81 +4,128 @@
+//                   new NewsData("ЯндексМузыка за 99 рублей!","https://yt3.ggpht.com/a/AGF-l7_3L_I5maGpVKDugzwEPEak9al3ufizsrkUZw=s900-c-k-c0xffffffff-no-rj-mo","Акция невиданной щедрости, аж в два раза дешевле!"),
+//                   new NewsData("K-POP Переоценен!","https://cdn.eksmo.ru/v2/ITD000000001043430/COVER/cover1__w600.jpg","Не слушал, но смею судить."),
+//                   new NewsData("Что стало с Т.А.Т.У.?","https://n1s1.hsmedia.ru/d4/f0/36/d4f03680616433d136e26a755a1aaa88/2143x1295_0x0a330ca2_13653060181557641052.jpeg","Никто не знает, но каждый переодически переслушивает их песни."),
+//                   new NewsData("Рецензия на клип: Вася Обломов - «Диги-диги»","https://cdn1.intermedia.ru/img/news/344756.jpg","Песня «Диги-диги» входила в недавний альбом Васи Обломова «Этот прекрасный мир» и была там одной из самых милых и мелодичных. Хотя «Диги-диги» и является сокращением от «дегенератов», в устах Васи это звучит почти нежно - что-то типа интонаций, с которой произносится  ...
```
remove commented code if it's unnecessary

tolmadima replied:
Deleted unnecessary part

ashvidchenko commented on app/src/main/java/com/example/lastfm/MainActivity.java:
```diff
@@ -4,81 +4,128 @@
+
     private void initRecyclerView() {
-        newsRecyclerView = findViewById(R.id.newsRecyclerView);
+        RecyclerView newsRecyclerView = findViewById(R.id.newsRecyclerView);
```
remove all deprecated usages of the word "news"

tolmadima replied:
Changed to Artist

ashvidchenko commented on app/src/main/java/com/example/lastfm/NewsAdapter.java:
```diff
@@ -35,46 +35,45 @@ public void clearItems(){
 
 
         class NewsViewHolder extends RecyclerView.ViewHolder  {
```
rename the class

tolmadima replied:
Changed whole class to ArtistAdapter and to ArtistsViewHolder

ashvidchenko commented on app/src/main/java/com/example/lastfm/Post.java:
```diff
@@ -0,0 +1,39 @@
+
+public class Post {
+    @SerializedName("name")
+    @Expose
```
What does the annotation mean?

tolmadima replied:
That class have no use right now. Deleted

ashvidchenko commented on app/src/main/java/com/example/lastfm/NewsData.java:
```diff
@@ -15,4 +15,9 @@
     public String getHeader(){return Header;}
     public String getImage(){return Image;}
     public String getDescription(){return Description;}
```
follow the code style and rename fields of the class

ashvidchenko commented on app/src/main/java/com/example/lastfm/RetrofitRequests.java:
```diff
@@ -0,0 +1,35 @@
+        }
+        return mInstance;
+    }
+    public interface JSONPlaceHolderApi {
```
rename it, u should use more meaningful name

tolmadima replied:
Deleted, interface have no use right now.

ashvidchenko commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -7,6 +7,13 @@
 
+    <android.widget.TextView
+        android:layout_width="match_parent"
+        android:layout_height="74dp"
```
create a constant

tolmadima replied:
There is no more TextView widget

ashvidchenko commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -7,6 +7,13 @@
+        android:layout_height="74dp"
+        tools:context="Getting Tags"
+        android:id="@+id/asynctask_view"
+        android:textSize="28sp"/>
```
create a constant

tolmadima replied:
Deleted

tolmadima committed:
Backup

tolmadima committed:
Не могу вывести данные.

Как я понял из onPostExetute я вытащить свой ArrayList не могу, но и описать вывод данных внутри этого метода у меня не получается. Вроде перечитал про асинктаск и про вывод в RecyclerView но все равно не допер, попробую еще завтра днем.

tolmadima committed:
Правки

ashvidchenko commented on app/src/main/java/com/example/lastfm/ArtistsData.java:
```diff
@@ -0,0 +1,22 @@
+package com.example.lastfm;
+
+public class ArtistsData {
```
u don't must always add suffix/prefix to class names, remove suffix "data"

ashvidchenko commented on app/src/main/java/com/example/lastfm/ArtistsData.java:
```diff
@@ -0,0 +1,22 @@
+public class ArtistsData {
+    private String ArtistName;
+    private String Image;
+    private String PlayCount;
```
rename fields according to the code style

tolmadima replied:
Changed for proper working with Gson

ashvidchenko commented on app/src/main/java/com/example/lastfm/MainActivity.java:
```diff
@@ -72,59 +71,44 @@ protected String doInBackground(String... strings) {
+                if (artist != null) {
+                    for (int i=0;i<artist.length();i++){
+                        listofartists.add(artist.getString(i));
+                        System.out.println(artist.getString(i));
```
read the link about logging in Android at last

tolmadima replied:
No more logging with println

ashvidchenko commented on app/src/main/java/com/example/lastfm/MainActivity.java:
```diff
@@ -72,59 +71,44 @@ protected String doInBackground(String... strings) {
                 e.printStackTrace();
             }
-            return artistName;
+            return listofartists;
```
use the camel case style

ashvidchenko commented on app/src/main/java/com/example/lastfm/MainActivity.java:
```diff
@@ -72,59 +71,44 @@ protected String doInBackground(String... strings) {
-//                   new NewsData("Что стало с Т.А.Т.У.?","https://n1s1.hsmedia.ru/d4/f0/36/d4f03680616433d136e26a755a1aaa88/2143x1295_0x0a330ca2_13653060181557641052.jpeg","Никто не знает, но каждый переодически переслушивает их песни."),
-//                   new NewsData("Рецензия на клип: Вася Обломов - «Диги-диги»","https://cdn1.intermedia.ru/img/news/344756.jpg","Песня «Диги-диги» входила в недавний альбом Васи Обломова «Этот прекрасный мир» и была там одной из самых милых и мелодичных. Хотя «Диги-диги» и является сокращением от «дегенератов», в устах Васи это звучит почти нежно - что-то типа интонаций, с которой произносится  ...
-           );
+        private ArrayList<ArtistsData> getArtists(){
```
u should use interface as the returned type instead of a realisation of this interface

tolmadima replied:
String deleted.

tolmadima committed:
Trying to make AsyncTask work

Added Parser.
Deleted NewsData.
Trying to parse data.

tolmadima committed:
Update MainActivity.java

tolmadima committed:
AsyncTask Completed.

tolmadima committed:
Deleted ImageView

ashvidchenko commented on app/src/main/java/com/example/lastfm/ArtistAdapter.java:
```diff
@@ -0,0 +1,87 @@
+public class ArtistAdapter extends RecyclerView.Adapter<ArtistAdapter.ArtistsViewHolder> {
+
+    private List<ArtistsData> artistsList = new ArrayList<>();
+
```
remove redundant empty lines everywhere

ashvidchenko commented on app/src/main/java/com/example/lastfm/ArtistAdapter.java:
```diff
@@ -0,0 +1,87 @@
+
+public class ArtistAdapter extends RecyclerView.Adapter<ArtistAdapter.ArtistsViewHolder> {
+
+    private List<ArtistsData> artistsList = new ArrayList<>();
```
rename it, this suffix is unneeded

ashvidchenko commented on app/src/main/java/com/example/lastfm/ArtistAdapter.java:
```diff
@@ -0,0 +1,87 @@
+
+
+
+    class ArtistsViewHolder extends RecyclerView.ViewHolder  {
```
create the separate file with this class

ashvidchenko commented on app/src/main/java/com/example/lastfm/ArtistsData.java:
```diff
@@ -0,0 +1,27 @@
+        this.playcount = artistPlayCount;
+    }
+
+    public String getArtistName(){return name;}
```
split lines

ashvidchenko commented on app/src/main/java/com/example/lastfm/MainActivity.java:
```diff
@@ -1,84 +1,108 @@
-    private RecyclerView newsRecyclerView;
-    private NewsAdapter newsAdapter;
+    private static final String APP_ID = "b4ab3bf82dcb495e182e04cfc1f12b7b";
+    private static final Integer NumberOfArtists = 10;
```
rename it, follow the code style

ashvidchenko commented on app/src/main/java/com/example/lastfm/MainActivity.java:
```diff
@@ -1,84 +1,108 @@
-        private void loadNews(){
-            Collection<NewsData> news = getNews();
-            newsAdapter.setItems(news);
+    private static class GetTopArtistTask extends AsyncTask<String, Void, List<ArtistsData>> {
```
create the separate file for this class

ashvidchenko commented on app/src/main/java/com/example/lastfm/MainActivity.java:
```diff
@@ -1,84 +1,108 @@
-            newsAdapter.setItems(news);
+    private static class GetTopArtistTask extends AsyncTask<String, Void, List<ArtistsData>> {
+
+        GetTopArtistTask(ArtistAdapter artistAdapter) { }
```
remove empty constructor

ashvidchenko commented on app/src/main/java/com/example/lastfm/MainActivity.java:
```diff
@@ -1,84 +1,108 @@
+
+        GetTopArtistTask(ArtistAdapter artistAdapter) { }
+        ArtistsData artistData = new ArtistsData();
+        List<ArtistsData> lsArtist = new ArrayList<>();
```
rename prefix

ashvidchenko commented on app/src/main/java/com/example/lastfm/MainActivity.java:
```diff
@@ -1,84 +1,108 @@
+    private static class GetTopArtistTask extends AsyncTask<String, Void, List<ArtistsData>> {
+
+        GetTopArtistTask(ArtistAdapter artistAdapter) { }
+        ArtistsData artistData = new ArtistsData();
```
why do u need it?

ashvidchenko commented on app/src/main/java/com/example/lastfm/MainActivity.java:
```diff
@@ -1,84 +1,108 @@
+                }
+
+                JSONObject topLevel = new JSONObject(builder.toString());
+                JSONObject main = topLevel.getJSONObject("artists");
```
use constants

ashvidchenko commented on app/src/main/java/com/example/lastfm/MainActivity.java:
```diff
@@ -1,84 +1,108 @@
-        newsRecyclerView.setAdapter(newsAdapter);
     }
 
+    public static void loadArtists(List<ArtistsData> artistData){
```
rename the method and its parameter

tolmadima committed:
Commit comments fix
